include/Neat/Reflection.h

```diff
@@ -122,7 +122,7 @@ namespace Neat
 		template<typename TObject, typename TType, TType TObject::* PtrToMember>
 		std::any get_value_erased(void* object)
 		{
-			TObject* object_ = reinterpret_cast<TObject*>(object);
+			TObject* object_ = static_cast<TObject*>(object);
```

Comment:
This is common practice to use `static_cast` instead of `reinterpret_cast` for `void*` (see https://stackoverflow.com/questions/310451/should-i-use-static-cast-or-reinterpret-cast-when-casting-a-void-to-whatever).

Reply:
Yeah I don't fully agree with that for various reasons, notably this implies that instead of reinterpret_casting a pointer, you should first convert it to void* and then static cast to avoid the reinterpret_cast.

But I'm not too attached to this, it should probably be changed to a std::any anyway.